dll/win32/browseui/addresseditbox.cpp

```diff
@@ -114,28 +182,17 @@ HRESULT STDMETHODCALLTYPE CAddressEditBox::ParseNow(long paramC)
         return hr;
 
     /* Get the path to browse and expand it if needed */
-    LPWSTR input;
-    int inputLength = fCombobox.GetWindowTextLength() + 2;
-
-    input = new WCHAR[inputLength];
-    fCombobox.GetWindowText(input, inputLength);
-
-    LPWSTR address;
-    int addressLength = ExpandEnvironmentStrings(input, NULL, 0);
-
-    if (addressLength <= 0)
+    CComHeapPtr<WCHAR> input, address;
+    if (!GetComboBoxText(input))
+        return E_FAIL;
+
+    INT addressLength = (wcschr(input, L'%') ? ::ExpandEnvironmentStrings(input, NULL, 0) : 0);
+    if (addressLength <= 0 ||
+        !address.Allocate(addressLength + 2) ||
```

Comment:
Allocate takes care of converting item size to byte size, so if this `2` is just `sizeof(WCHAR)` you can change it to 1.

Reply:
Fixed.